### PR TITLE
feat: Add WhatsApp Messaging channel

### PR DESCRIPTION
**Pull request by reinlabares**
Added documentation for Whispir's WhatsApp Messaging feature
Added WhatsApp Model and sub models

Committed by Whispir-Softwares on behalf of rlabares@whispir.com

### CODE REVIEW

**Comment by reinlabares** [comment]
@aleksandar-milisic @richardfontein 

This is now ready for re-review. 

Note: This is a new branch and a new PR. I have already applied the necessary changes that @richardfontein commented on the previous PR.

Thank you so much for the hlep

**Review by aleksandar-milisic — APPROVED** [review]

File: openapi.yaml [hunk 1]
```diff
@@ -10615,6 +10618,300 @@ components:
...
+                - text: 'https://hsbc.com/renewal'
+                - text: 'https://hsbc.com/terms'
+                - text: 'https://hsbc.com/privacy'
+      properties:
+        type:
+          type: string
+          description: |-
+            The type of WhatsApp message to be sent, e.g. "template".
+
+            Whispir currently only supports type "template".
```
Comment (richardfontein) [inline openapi.yaml:10770]: If only a set of values can be taken, we can use the JSON schema [enum](https://json-schema.org/understanding-json-schema/reference/generic.html#enumerated-values) key like follows:

```
type: string
enum:
  - template
```

File: openapi.yaml [hunk 2]
```diff
@@ -10615,6 +10618,300 @@ components:
         - errorText
         - errorDetail
       title: API Error
+    WhatsApp:
```
Comment (richardfontein) [inline openapi.yaml:10621]: Please tag this schema with `Messages` to have it displayed in the Message API group:
```
x-tags:
  - Messages
```

File: openapi.yaml [hunk 3]
```diff
@@ -10615,6 +10618,300 @@ components:
...
+      description: 'The content of your message which may contain any text and media that will be sent to the recipient. The contents can include a header, a body, and buttons.'
+      properties:
+        header:
+          $ref: '#/components/schemas/WAContentHeader'
+        body:
+          type: array
+          description: The contents of your WhatsApp message
+          items:
+            title: Object
+            type: object
```
Comment (richardfontein) [inline openapi.yaml:10798]: What are the key/value pairs required for this object? The `object` type doesn't make it clear to developers what is required, we need to provide the `properties` to specify the requirements here.

File: openapi.yaml [hunk 4]
```diff
@@ -10615,6 +10618,300 @@ components:
...
+              payload: 'https://google.com'
+      properties:
+        index:
+          type: string
+          description: The index value of the button this object represents
+          example: '0'
+        sub_type:
+          type: string
+          description: The type of dynamic button i.e. "quick reply" or "url"
+          example: 'url'
```
Comment (richardfontein) [inline openapi.yaml:10899]: We can also use an `enum` for this property:

```
type: string
enum:
  - quick_reply
  - url
```

File: openapi.yaml [hunk 5]
```diff
@@ -10615,6 +10618,300 @@ components:
...
+          title: Object
+          type: object
+          description: Link to video location
+        document:
+          title: Object
+          type: object
+          description: Link to document location
+        location:
+          title: Object
+          type: object
```
Comment (richardfontein) [inline openapi.yaml:10867]: What are the key-value pairs required for these objects:

* `image`
* `video`
* `document`
* `location`

Could we please add the appropriate `properties` for each of these.

**Review by richardfontein — CHANGES_REQUESTED** [review]

File: openapi.yaml (hunk 1 above)
Comment (reinlabares) [reply openapi.yaml:10774]: Added enum to type property

File: openapi.yaml (hunk 2 above)
Comment (reinlabares) [reply openapi.yaml:10621]: Added x-tags to WhatsApp

File: openapi.yaml (hunk 3 above)
Comment (reinlabares) [reply openapi.yaml:10802]: Added properties to the body object

File: openapi.yaml (hunk 4 above)
Comment (reinlabares) [reply openapi.yaml:10943]: Added enum to button subtype

File: openapi.yaml (hunk 5 above)
Comment (reinlabares) [reply openapi.yaml:10891]: added properties to the WAContentHeader objects

File: openapi.yaml [hunk 6]
```diff
@@ -10615,6 +10618,344 @@ components:
...
+              payload: 'https://google.com'
+      properties:
+        index:
+          type: string
+          description: The index value of the button this object represents
+          example: '0'
+        sub_type:
+          type: string
+          enum:
+            - quick reply
```
Comment (richardfontein) [inline openapi.yaml:10940]: hey @reinlabares , there's a small discrepancy between these two lines, one includes an underscrore `"_"` delimiter and the other is space `" "` delimitered. Could you please align for consistency
* https://github.com/whispir/openapi/pull/200/files#diff-d910ba2ef878f7db0223a966b81c8b3f3b65027bb39e4431bb05140171eece39R10927
* https://github.com/whispir/openapi/pull/200/files#diff-d910ba2ef878f7db0223a966b81c8b3f3b65027bb39e4431bb05140171eece39R10940

**Review by richardfontein — CHANGES_REQUESTED** [review]

File: openapi.yaml (hunk 6 above)
Comment (reinlabares) [reply openapi.yaml]: Good catch! Thank you. I have addressed this in the latest commit

**Review by richardfontein — APPROVED** [review]
LGTM! Thanks again for your contribution

**Pull request merged**